src/basic/escape.c

```diff
@@ -237,8 +237,8 @@ int cunescape_one(const char *p, size_t length, char *ret, uint32_t *ret_unicode
                                 return a[i];
                 }
 
-                c = ((uint32_t) a[0] << 28U) | ((uint32_t) a[1] << 24U) | ((uint32_t) a[2] << 20U) | ((uint32_t) a[3] << 16U) |
-                    ((uint32_t) a[4] << 12U) | ((uint32_t) a[5] <<  8U) | ((uint32_t) a[6] <<  4U) |  (uint32_t) a[7];
+                c = (a[0] << 28U) | (a[1] << 24U) | (a[2] << 20U) | (a[3] << 16U) |
+                    (a[4] << 12U) | (a[5] <<  8U) | (a[6] <<  4U) | (a[7] <<  0U);
```

Comment:
Is this really the case? i mean, how is the resulting type determined if you use where an int is on the left side, and an unsigned int on the right side? to my knowledge the left side is what defines the result, and since that is an int, the result will be an int too. 15 << 28U is -268435456 hence.